### PR TITLE
Add downloader for LANUV measurements

### PR DESCRIPTION
**Pull request by jdk24**

### CODE REVIEW

File: build-images.sh [hunk 1]
```diff
@@ -27,6 +27,12 @@ docker tag sauberprojekt/raster_download:$MASTER_TAG sauberprojekt/raster_downlo
 docker build --rm -f "json_download/Dockerfile" -t sauberprojekt/json_download:$MASTER_TAG "json_download"
 docker tag sauberprojekt/json_download:$MASTER_TAG sauberprojekt/json_download:$DATE_TAG
 
+docker build --rm -f "lanuv_download/Dockerfile" -t sauberprojekt/lanuv_download:$MASTER_TAG "lanuv_download"
+docker tag sauberprojekt/lanuv_download:$MASTER_TAG sauberprojekt/lanuv_download:$DATE_TAG
+
```
Comment (JakobMiksch) [inline build-images.sh:32]: the order of `lanuv` and `lubw` is opposite in above
would better to keep the same order for  consistency

File: lanuv_download/resources/scripts/dl_lanuv.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	fi
+done
+
+OUTDIR=/lanuv_data
+OUTFILE=lanuv_$(TZ=Europe/Berlin date +%Y%m%d%H%M).csv
+
+# Download xml file from server. Exit if curl throws failure (http response!=2xx)
+if curl -s --fail "https://www.lanuv.nrw.de/fileadmin/lanuv/luft/immissionen/aktluftqual/eu_luftqualitaet.csv" -o $OUTDIR/$OUTFILE;
+then
+    echo "Datei $OUTDIR/$OUTFILE heruntergeladen."
```
Comment (JakobMiksch) [inline lanuv_download/resources/scripts/dl_lanuv.sh:52]: text should be in English unless there is a good reason for it

File: lanuv_download/resources/scripts/dl_lanuv.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+else
+    echo "Error: Datei konnte nicht herunterladen werden."
```
Comment (JakobMiksch) [inline lanuv_download/resources/scripts/dl_lanuv.sh:54]: text should be in English

File: lanuv_download/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
+FROM alpine:latest 
+
+# install dependencies
+RUN apk update && \
+    apk del sed && \
```
Comment (JakobMiksch) [inline lanuv_download/Dockerfile:5]: why is `sed` removed and then installed again?

**Review by JakobMiksch — COMMENTED** [review]
Looks good so far. I added some comments.
I still need to test it locally

File: lanuv_download/resources/scripts/dl_lanuv.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -e 
+
+file_env() {
```
Comment (JakobMiksch) [inline lanuv_download/resources/scripts/dl_lanuv.sh:5]: documentation what this function does would be  helpful

**Comment by JakobMiksch** [comment]
I ran the  script below inside the docker container and got this error. 
Is the error on my side  or is there something  wrong with the  script?

```shell
bash-5.1# ./dl_lanuv.sh                                                    
Datei /lanuv_data/lanuv_202103021346.csv heruntergeladen.                  
ERROR:  column "offset_hrs" does not exist                                 
LINE 26:     ON CONFLICT (date_time, fk_component, fk_station, offset...   
                         ^                                                 
QUERY:  WITH                                                               
      lut_stat AS                                                          
      (SELECT * FROM  station_data.lut_station)                            
      ,                                                                    
      lut_comp AS                                                          
      (SELECT * FROM  station_data.lut_component)                          
                                                                           
  INSERT INTO                                                              
      station_data.tab_measurement                                         
    (                                                                      
      val,                                                                 
      date_time,                                                           
      fk_component,                                                        
      fk_station                                                           
    )                                                                      
                                                                           
    SELECT                                                                 
      tmp.val,                                                             
      tmp.date_time,                                                       
      lut_comp.idpk_component,                                             
      lut_stat.idpk_station                                                
                                                                           
    FROM tmp                                                               
    JOIN lut_stat ON tmp.station = lut_stat.station_code                   
    JOIN lut_comp on tmp.component = lut_comp.component_name               
    ON CONFLICT (date_time, fk_component, fk_station, offset_hrs)          
    DO NOTHING                                                             
CONTEXT:  PL/pgSQL function station_data.lanuv_parse(text) line 67 at SQL s
tatement                                                                   
```

File: lanuv_download/Dockerfile (hunk 4 above)
Comment (jdk24) [reply lanuv_download/Dockerfile:5]: Alpine docker base image comes with some limited busybox-sed-implementation. Remove and install fully-fledged sed package for parser function.

**Comment by jdk24** [comment]
> Is the error on my side or is there something wrong with the script?

Good point. I'm currently adjusting the database init sql files to also reflect recent changes and additions. 
This is one such example. 
Since #64 requires a DB restart, I suggest appending  those changes there.

File: lanuv_download/Dockerfile (hunk 4 above)
Comment (JakobMiksch) [reply lanuv_download/Dockerfile:5]: alright,  I did not know that

**Comment by chrismayer** [comment]
Is it intended that the `lubw_downloader` is commented out?

**Comment by jdk24** [comment]
The project's own LUBW endpoint has not been available recently. Just checked again locally.

If not available soon, possibly query and parse from data LUBW's public API (same / more info).
Done here: https://github.com/jdk24/fossgis_project/tree/master/project/scripts 

Will add this question to the agenda.

Meanwhile: Leave in yaml as commented-out or remove entirely?

**Comment by chrismayer** [comment]
> Meanwhile: Leave in yaml as commented-out or remove entirely?

Leave in yaml as commented-out is fine. Question was more for understanding / clarification.

**Review by chrismayer — APPROVED** [review]
LGTM, thanks! Please merge at will.

**Pull request merged**